From cecee90833998c7e25acdae4ff8e0dfdab8f4bf0 Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Mon, 3 Sep 2018 19:11:45 +0300
Subject: [PATCH 51/53] ARM: imx7d: cl-som-imx7: dts: add LVDS support

MIPI DSI to LVDS bridge configuration for LVDS panel
Ampire 7"-AM-1280800P

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 arch/arm/boot/dts/Makefile                |  1 +
 arch/arm/boot/dts/imx7d-sbc-imx7-lvds.dts | 67 +++++++++++++++++++++++++++++++
 2 files changed, 68 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx7d-sbc-imx7-lvds.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 81d5bf2..7c54a83 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -548,6 +548,7 @@ dtb-$(CONFIG_SOC_IMX7D) += \
 	imx7d-colibri-eval-v3.dtb \
 	imx7d-nitrogen7.dtb \
 	imx7d-sbc-imx7.dtb \
+	imx7d-sbc-imx7-lvds.dtb \
 	imx7d-sbc-imx7-m4.dtb \
 	imx7d-sbc-iot-imx7.dtb \
 	imx7d-sbc-iot-imx7-can.dtb \
diff --git a/arch/arm/boot/dts/imx7d-sbc-imx7-lvds.dts b/arch/arm/boot/dts/imx7d-sbc-imx7-lvds.dts
new file mode 100644
index 0000000..47d3e72
--- /dev/null
+++ b/arch/arm/boot/dts/imx7d-sbc-imx7-lvds.dts
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2017 CompuLab Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx7d-sbc-imx7.dts"
+
+/ {
+	mipi_dsi_reset: mipi-dsi-reset {
+		compatible = "gpio-reset";
+		reset-gpios = <&gpio1 8 GPIO_ACTIVE_LOW>; /* Unused gpio - routed to P4_21 */
+		reset-delay-us = <100>;
+		#reset-cells = <0>;
+	};
+
+	regulators {
+		reg_mipi_dsi_pwr_on: mipi_dsi_pwr_on {
+			compatible = "regulator-fixed";
+			regulator-name = "mipi_dsi_pwr_on";
+			gpio = <&gpio1 9 GPIO_ACTIVE_HIGH>;  /* Unused gpio - routed to P4_23 */
+			enable-active-high;
+		};
+	};
+};
+
+&lcdif {
+	disp-dev = "mipi_dsi_samsung";
+};
+
+&mipi_dsi {
+	lcd_panel = "SN65DSI_default";
+	disp-power-on-supply = <&reg_mipi_dsi_pwr_on>;
+	resets = <&mipi_dsi_reset>;
+	status = "okay";
+};
+
+&i2c2 {
+	ti_bridge: sn65dsi83@2c {
+		compatible = "ti,sn65dsi83";
+		reg = <0x2c>;
+		enable-gpios = <&pca9555 2 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+		lvds_panel {
+			/* Ampire 7"-AM-1280800P */
+			lvds-bpp = <24>;
+			lvds-format = <0>;
+			display-timings {
+				clock-frequency = <62500000>;
+				hactive = <1280>;
+				vactive = <800>;
+				hback-porch = <6>;
+				hfront-porch = <5>;
+				vback-porch = <2>;
+		                vfront-porch = <3>;
+			        hsync-len = <2>;
+				vsync-len = <1>;
+				hsync-active = <0>;
+			        vsync-active = <0>;
+				de-active = <0>;
+			        pixelclk-active = <0>;
+			};
+		};
+	};
+};
-- 
2.7.4

